content/documentation/pages/6-recipes/4-kafka/1-ext-kafka-cluster-cf.md

```diff
@@ -0,0 +1,146 @@
+
+## Prerequisite
+
+Let's start with preparing the external Kafka cluster credentials.
+
+Typically, a series of Kafka brokers, collectively are referred to as a Kafka cluster.
+The brokers individually can be reached through its external IP addresses or a well-defined DNS route could be available to use as well.
+
+For this walk-through, though, we will stick to a simpler setup of 3-broker cluster with their DNS addresses being `foo0.broker.foo`, `foo1.broker.foo`, and `foo2.broker.foo`. The default port of the brokers is `9092`.
+
+If the cluster is secured, depending on the security option in use at the broker, different properties are expected to be supplied for when applications attempt to connect to the external cluster.
+Again, for simplicity, we will use `PlainLoginModule` with username as `test` and password as `bestest`.
+
+## User-provided Services vs. Spring Boot Properties
+
+The next question Cloud Foundry developers stumble upon is whether or not to set up Kafka connection as Cloud Foundry custom user-provided service (CUPS) or simply passing connection credentials as Spring Boot properties.
+
+In Cloud Foundry, there isn't a Spring Cloud Connector or CF JavaEnv support for Kafka, so by service-binding the Kafka CUPS with the application, you _will not_ automatically be able to parse `VCAP_SERVICES` and pass the connection credentials to the applications dynamically.
+Even with CUPS in place, it is your responsibility to parse the `VCAP_SERVICES` JSON and pass them as Boot properties, so no real benefit.
+You can see an example of CUPS in action in the Spring Cloud Data Flow's reference guide.
+
+For this walk-through, we will stick to the Spring Boot properties.
+
+## Standalone Streaming Apps
+
+The typical Cloud Foundry deployment of an application includes a `manifest.yml` file.
+We will use the `source-sample` source application to highlight the configurations to connect to an external Kafka cluster.
+
+```yaml
+---
+applications:
+- name: source-sample
+  host: source-sample
+  memory: 1G
+  disk_quota: 1G
+  instances: 1
+  path: source-sample.jar
+env:
+    SPRING_APPLICATION_JSON: '{"spring.cloud.stream.kafka.binder.brokers":"foo0.broker.foo,foo1.broker.foo,foo2.broker.foo","spring.spring.cloud.stream.kafka.binder.jaas.options.username":"test","spring.spring.cloud.stream.kafka.binder.jaas.options.password":"bestest","spring.spring.cloud.stream.kafka.binder.jaas.loginModule":"org.apache.kafka.common.security.plain.PlainLoginModule",
+    "spring.cloud.stream.bindings.output.destination":"fooTopic"}'
+```
+
+With the above setting, when the `source-sample` source is deployed to Cloud Foundry, it should be able to connect to the external cluster.
+
+You can verify the connection credentials by accessing `source-sample`'s `/configprops` actuator endpoint.
+Likewise, you will also see the connection credentials printed in the app logs.
+
+[[note]]
```

Comment:
I think there is still value in extracting the actual credentials from the manifest.yml/application.properties into CUPS and then reference only the VCAP variable (even though that is hardcoded).